🐙 source-sendgrid: run up-to-date pipeline [2025-01-11]

airbyteio opened the pull request:
# Update source-sendgrid

This PR was autogenerated by running `airbyte-ci connectors --name=source-sendgrid up_to_date --pull`

We've set the `auto-merge` label on it, so it will be automatically merged if the CI pipelines pass.
If you don't want to merge it automatically, please remove the `auto-merge` label.
Please reach out to the Airbyte Connector Tooling team if you have any questions or concerns.


## Operations

- Upgrade the base image to the latest version in metadata.yaml: Successful

- PATCH bump source-sendgrid version to 1.2.3: Successful

- Build source-sendgrid docker image for platform(s) linux/amd64, linux/arm64: Successful

- Get dependency updates: Successful

- Create or update pull request on Airbyte repository: Successful

- Add changelog entry: Successful




## Dependency updates

We use [`syft`](https://github.com/anchore/syft) to generate a SBOM for the latest connector version and the one from the PR.
It allows us to spot the dependencies that have been updated at all levels and for all types of dependencies (system, python, java etc.).
Here are the dependencies that have been updated compared to `airbyte/source-sendgrid:latest`.
Keep in mind that `:latest` does not always match the connector code on the main branch. 
It is the latest released connector image when the head commit of this branch was created.

| Type | Name | State | Previous Version | New Version |
|------|------|-------|-------------|------------------|
| python | Jinja2 | updated | 3.1.4 | **3.1.5** |
| python | MarkupSafe | updated | 2.1.5 | **3.0.2** |
| python | PyJWT | updated | 2.9.0 | **2.10.1** |
| python | PyYAML | updated | 6.0.1 | **6.0.2** |
| python | airbyte-cdk | updated | 5.17.0 | **6.17.0** |
| python | airbyte_protocol_models_dataclasses | updated | 0.13.0 | **0.14.1** |
| python | attrs | updated | 23.2.0 | **24.2.0** |
| deb | base-files | updated | 12.4+deb12u6 | **12.4+deb12u8** |
| python | bracex | updated | 2.4 | **2.5.post1** |
| deb | bsdutils | updated | 1:2.38.1-5+deb12u1 | **1:2.38.1-5+deb12u2** |
| python | cachetools | updated | 5.4.0 | **5.5.0** |
| python | cattrs | updated | 23.2.3 | **24.1.2** |
| python | certifi | updated | 2024.7.4 | **2024.12.14** |
| python | cffi | updated | 1.16.0 | **1.17.1** |
| python | charset-normalizer | updated | 3.3.2 | **3.4.0** |
| python | click | updated | 8.1.7 | **8.1.8** |
| python | cryptography | updated | 42.0.8 | **43.0.3** |
| python | distlib | updated | 0.3.8 | **0.3.9** |
| python | filelock | updated | 3.15.4 | **3.16.1** |
| python | genson | updated | 1.2.2 | **1.3.0** |
| python | idna | updated | 3.7 | **3.10** |
| python | langsmith | updated | 0.1.96 | **0.1.147** |
| deb | libarchive13 | updated | 3.6.2-1+deb12u1 | **3.6.2-1+deb12u2** |
| deb | libblkid1 | updated | 2.38.1-5+deb12u1 | **2.38.1-5+deb12u2** |
| deb | libc-bin | updated | 2.36-9+deb12u7 | **2.36-9+deb12u9** |
| deb | libc6 | updated | 2.36-9+deb12u7 | **2.36-9+deb12u9** |
| deb | libcurl4 | updated | 7.88.1-10+deb12u6 | **7.88.1-10+deb12u8** |
| deb | libexpat1 | updated | 2.5.0-1 | **2.5.0-1+deb12u1** |
| deb | libglib2.0-0 | updated | 2.74.6-2+deb12u3 | **2.74.6-2+deb12u4** |
| deb | libglib2.0-data | updated | 2.74.6-2+deb12u3 | **2.74.6-2+deb12u4** |
| deb | libmount1 | updated | 2.38.1-5+deb12u1 | **2.38.1-5+deb12u2** |
| deb | libnghttp2-14 | updated | 1.52.0-1+deb12u1 | **1.52.0-1+deb12u2** |
| deb | libnss3 | updated | 2:3.87.1-1 | **2:3.87.1-1+deb12u1** |
| deb | libsmartcols1 | updated | 2.38.1-5+deb12u1 | **2.38.1-5+deb12u2** |
| deb | libsqlite3-0 | updated | 3.40.1-2 | **3.40.1-2+deb12u1** |
| deb | libssl3 | updated | 3.0.13-1~deb12u1 | **3.0.15-1~deb12u1** |
| deb | libsystemd0 | updated | 252.26-1~deb12u2 | **252.31-1~deb12u1** |
| deb | libudev1 | updated | 252.26-1~deb12u2 | **252.31-1~deb12u1** |
| deb | libuuid1 | updated | 2.38.1-5+deb12u1 | **2.38.1-5+deb12u2** |
| python | more-itertools | updated | 10.3.0 | **10.5.0** |
| deb | mount | updated | 2.38.1-5+deb12u1 | **2.38.1-5+deb12u2** |
| python | msgpack | updated | 1.0.8 | **1.1.0** |
| python | nltk | updated | 3.8.1 | **3.9.1** |
| deb | openssl | updated | 3.0.13-1~deb12u1 | **3.0.15-1~deb12u1** |
| python | orjson | updated | 3.10.7 | **3.10.14** |
| python | packaging | updated | 24.1 | **23.2** |
| python | pkginfo | updated | 1.11.1 | **1.11.2** |
| python | platformdirs | updated | 4.2.2 | **3.11.0** |
| python | pydantic | updated | 2.8.2 | **2.10.5** |
| python | pydantic_core | updated | 2.20.1 | **2.27.2** |
| python | pyproject_hooks | updated | 1.1.0 | **1.2.0** |
| python | pytz | updated | 2024.1 | **2024.2** |
| python | regex | updated | 2024.7.24 | **2024.11.6** |
| python | serpyco-rs | updated | 1.10.2 | **1.11.0** |
| python | setuptools | updated | 72.1.0 | **70.0.0** |
| python | six | updated | 1.16.0 | **1.17.0** |
| python | tomli | updated | 2.0.1 | **2.1.0** |
| python | tomlkit | updated | 0.12.5 | **0.13.2** |
| python | tqdm | updated | 4.66.5 | **4.67.1** |
| python | trove-classifiers | updated | 2024.7.2 | **2024.10.21.16** |
| python | tzdata | updated | 2024.1 | **2024.2** |
| python | urllib3 | updated | 2.2.2 | **2.2.3** |
| deb | util-linux | updated | 2.38.1-5+deb12u1 | **2.38.1-5+deb12u2** |
| deb | util-linux-extra | updated | 2.38.1-5+deb12u1 | **2.38.1-5+deb12u2** |
| python | virtualenv | updated | 20.26.3 | **20.27.1** |
| python | wcmatch | updated | 8.4 | **10.0** |
| python | wheel | updated | 0.43.0 | **0.44.0** |
| python | RapidFuzz | added | not present | **3.10.1** |
| python | Unidecode | added | not present | **1.3.8** |
| python | anyio | added | not present | **4.8.0** |
| python | dunamai | added | not present | **1.23.0** |
| python | h11 | added | not present | **0.14.0** |
| python | httpcore | added | not present | **1.0.7** |
| python | httpx | added | not present | **0.28.1** |
| python | psutil | added | not present | **6.1.0** |
| python | python-ulid | added | not present | **3.0.0** |
| python | sniffio | added | not present | **1.3.1** |
| python | Deprecated | removed | 1.2.14 | **not present** |
| python | autocommand | removed | 2.2.2 | **not present** |
| python | backports.tarfile | removed | 1.2.0 | **not present** |
| python | importlib_metadata | removed | 8.0.0 | **not present** |
| python | importlib_resources | removed | 6.4.0 | **not present** |
| python | inflect | removed | 7.3.1 | **not present** |
| python | jaraco.context | removed | 5.3.0 | **not present** |
| python | jaraco.functools | removed | 4.0.1 | **not present** |
| python | jaraco.text | removed | 3.12.1 | **not present** |
| python | my-test-package | removed | 1.0 | **not present** |
| python | ordered-set | removed | 4.1.0 | **not present** |
| python | rapidfuzz | removed | 3.9.4 | **not present** |
| python | typeguard | removed | 4.3.0 | **not present** |
| python | wrapt | removed | 1.16.0 | **not present** |
| python | zipp | removed | 3.19.2 | **not present** |

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=48238) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=48238) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=48238) it.</sub>

airbyteio committed:
Upgrade the base image to the latest version in metadata.yaml
PATCH bump source-sendgrid version to 1.2.3
Build source-sendgrid docker image for platform(s) linux/amd64, linux/arm64
Get dependency updates
Create or update pull request on Airbyte repository
Add changelog entry

Commit:
Fix upgrade to concurrency

maxi297 commented:
Started some regression tests here: https://github.com/airbytehq/airbyte/actions/runs/12789626639

Commit:
Merge branch 'master' into up-to-date/source-sendgrid

maxi297 commented:
CATs errors:
* The state now has a string version because it has been moved to concurrency

Regression testing changes:
* is_resumable is not false for full-refresh streams that do pagination
* mitmproxy does not like encoded responses so connection with the contacts stream have failed
* All the full_refresh streams haven't returned any record. I can't reproduce locally. I'll try a run of the regression tests without the state...

maxi297 commented:
Running the full-refresh streams without state [here](https://github.com/airbytehq/airbyte/actions/runs/12791595801/job/35660072913), it seems like everything is fine. I've tested the contacts stream locally to bypass the `mitmproxy` issue. Assuming the incremental streams passed (I'll do more local tests), we should be good

Commit:
remove abnormal state test

maxi297 commented:
I tested some of the streams with state like the following and I can get only the records that are expected:
```
[
  {
    "type": "STREAM",
    "stream": {
      "stream_state": {
        "created": 1736959369
      },
      "stream_descriptor": {
        "name": "blocks"
      }
    }
  }
]
```
